modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_2/DDMFormInstanceUpgradeProcess.java

```diff
@@ -0,0 +1,236 @@
+			}
+		}
+	}
+
+	private void _upgradeDDMFormInstanceVersion(
+			Date expirationDate, long formInstanceId,
+			PreparedStatement selectPreparedStatement,
+			PreparedStatement updatePreparedStatement)
+		throws Exception {
+
+		selectPreparedStatement.setLong(1, formInstanceId);
+
+		try (ResultSet resultSet = selectPreparedStatement.executeQuery()) {
+			while (resultSet.next()) {
+				String settings = resultSet.getString("settings_");
+
+				if (Validator.isNotNull(settings)) {
+					JSONObject settingsJSONObject =
+						_jsonFactory.createJSONObject(settings);
+
+					_upgradeSettings(
+						expirationDate,
+						settingsJSONObject.getJSONArray("fieldValues"));
+
+					updatePreparedStatement.setString(
+						1, settingsJSONObject.toJSONString());
+
+					updatePreparedStatement.setLong(
+						2, resultSet.getLong("formInstanceVersionId"));
+
+					updatePreparedStatement.addBatch();
+				}
+			}
+		}
+	}
+
+	private void _upgradeSettings(
+		Date expirationDate, JSONArray fieldValuesJSONArray) {
+
+		boolean hasExpirationDate = false;
+		boolean hasLimitToOneSubmissionPerUser = false;
+		boolean hasNeverExpire = false;
+		boolean hasShowPartialResultsToRespondents = false;
+
+		for (int i = 0; i < fieldValuesJSONArray.length(); i++) {
+			JSONObject fieldValueJSONObject =
+				fieldValuesJSONArray.getJSONObject(i);
+
+			if (Objects.equals(
+					fieldValueJSONObject.getString("name"), "expirationDate")) {
+
+				fieldValueJSONObject.put("value", _getString(expirationDate));
+
+				hasExpirationDate = true;
+			}
+			else if (Objects.equals(
+						fieldValueJSONObject.getString("name"),
+						"limitToOneSubmissionPerUser")) {
+
+				fieldValueJSONObject.put(
+					"value", String.valueOf(expirationDate != null));
+
+				hasLimitToOneSubmissionPerUser = true;
+			}
+			else if (Objects.equals(
+						fieldValueJSONObject.getString("name"),
+						"neverExpire")) {
+
+				fieldValueJSONObject.put(
+					"value", String.valueOf(expirationDate == null));
+
+				hasNeverExpire = true;
+			}
+			else if (Objects.equals(
+						fieldValueJSONObject.getString("name"),
+						"showPartialResultsToRespondents")) {
+
+				fieldValueJSONObject.put(
+					"value", String.valueOf(expirationDate != null));
+
+				hasShowPartialResultsToRespondents = true;
+			}
+		}
+
+		if (!hasExpirationDate) {
+			fieldValuesJSONArray.put(
+				_createFieldValue(
+					"expirationDate", _getString(expirationDate)));
+		}
+
+		if (!hasLimitToOneSubmissionPerUser) {
+			fieldValuesJSONArray.put(
+				_createFieldValue(
+					"limitToOneSubmissionPerUser",
+					String.valueOf(expirationDate != null)));
+		}
+
+		if (!hasNeverExpire) {
+			fieldValuesJSONArray.put(
+				_createFieldValue(
+					"neverExpire", String.valueOf(expirationDate == null)));
+		}
+
+		if (!hasShowPartialResultsToRespondents) {
+			fieldValuesJSONArray.put(
+				_createFieldValue(
+					"showPartialResultsToRespondents",
+					String.valueOf(expirationDate != null)));
+		}
+	}
```

Comment:
@rodrigopaulino, can you please give more details why the value of `` is being set according the `expirationDate` actual value?

I apologize to you, but I didn't get the reason. :grin:

Reply:
@natocesarrego, if it contains expiration date, it means that it used to be a Poll, so we'll turn this setting on to show the partial results, as it was with the Polls product.